keps/sig-apps/592-ttl-after-finish/README.md

```diff
@@ -250,17 +219,200 @@ Mitigations:
 
 ## Graduation Criteria
 
-We want to implement this feature for Pods/Jobs first to gather feedback, and
-decide whether to generalize it to custom resources. This feature can be
-promoted to beta after we finalize the decision for whether to generalize it or
-not, and when it satisfies users' need for cleaning up finished resource
-objects, without regressions.
+### Alpha
+
+ - For alpha graduation, the feature implemented for Job, as future work it can be extended to Pods, but that should happen under a separate feature flag.
+ - Unit and e2e tests
+
+### Alpha -> Beta
+
+- Appropriate metrics are agreed on and implemented
+- upgrade/rollback manually tested 
 
-This will be promoted to GA once it's gone a sufficient amount of time as beta
-with no changes. 
+### Beta -> GA
+
+- Make a decision on wehther or not the feature should be extended to pods
+- Enabled in Beta for at least two releases without complaints
 
 [umbrella issues]: https://github.com/kubernetes/kubernetes/issues/42752
 
-## Implementation History
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: TTLAfterFinished
+    - Components depending on the feature gate: kube-apiserver, kube-controller-manager
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+  No.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes. One caveat here is that Jobs created with TTLSecondsAfterFinished set when 
+  the feature was enabled will continue to have that field set when the feature is disabled,
+  but will not have any effect.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  It should work as expected.
+
+* **Are there any tests for feature enablement/disablement?**
+  No.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  It shouldn't impact already running workloads. This is an opt-in feature since users need to 
+  explicitly set the TTLSecondsAfterFinished parameter in the job spec, which is silently dropped
+  by the api-server if the feature is disabled.
```

Comment:
The field isn't dropped, right?
We're preserving the fields if they were set.

Reply:
It is dropped: https://github.com/kubernetes/kubernetes/blob/d72c056260e771e8cd01220203582de0c0015786/pkg/registry/batch/job/strategy.go#L79

Reply:
Hmm - which isn't really a pattern we want to have. We generally want to avoid clearing explicitly set fields, e.g. in:
https://github.com/kubernetes/kubernetes/blob/d72c056260e771e8cd01220203582de0c0015786/pkg/api/pod/util.go#L450

@liggitt - FYI

Reply:
This is an existing behavior, do we want to change it?

Reply:
I think we want - though please reach out to @liggitt (or other API approver) for final confirmation.

Reply:
The current behavior looks correct.. if the feature is disabled, we only preserve the field if it was already set in the **persisted** object, and clear it otherwise. That

Reply:
done.